presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -813,7 +814,11 @@ public PlanOptimizers(
                         costCalculator,
                         ImmutableSet.of(new ScaledWriterRule())));
 
-        if (!forceSingleNode) {
+        if (featuresConfig.isNativeExecutionEnabled() && featuresConfig.isNativeSingleWorkerExecution()) {
```

Comment:
But for some cases, it would be exchange from coordinator to worker? For example, scanning system table
Aggregation [Worker]
|
Exchange
|
TableScan (system table) [Coordinator]